mmcv/ops/csrc/onnxruntime/cpu/corner_pool.cpp

```diff
@@ -0,0 +1,115 @@
+                } // for h
+            }  // for w
+        } // for c
+    }  // for n
+
+}
+
+void LeftPoolForwardCPU(const float *input, float *output,
+                          const int nthreads, const int channels,
+                          const int height, const int width) {
+    int batch_size = nthreads / channels / width / height;
+    for (int n = 0; n < batch_size; n++) {
+        int index_n = n * channels * width * height;
+        for (int c = 0; c < channels; c++) {
+            int index_n_c = index_n + c * width * height;
+            for (int h = 0; h < height; h++) {
+                // directly copy the most right value from input to output
+                output[index_n_c + h * width + width - 1] = input[index_n_c + h * width + width - 1];
+                // do left_pool
+                for (int w = width - 2; w >= 0; w--){
+                    output[index_n_c + h * width + w] = std::max(output[index_n_c + h * width + w + 1], input[index_n_c + h * width +w]);
+                } // for w
+            }  // for h
+        } // for c
+    }  // for n
+
+}
+
+void RightPoolForwardCPU(const float *input, float *output,
+                          const int nthreads, const int channels,
+                          const int height, const int width) {
+    int batch_size = nthreads / channels / width / height;
+    for (int n = 0; n < batch_size; n++) {
+        int index_n = n * channels * width * height;
+        for (int c = 0; c < channels; c++) {
+            int index_n_c = index_n + c * width * height;
+            for (int h = 0; h < height; h++) {
+                // directly copy the most left value from input to output
+                output[index_n_c + h * width] = input[index_n_c + h * width];
+                // do right_pool
+                for (int w = 1; w < width; w++) {
+                    output[index_n_c + h * width + w] = std::max(output[index_n_c + h * width + w - 1], input[index_n_c + h * width + w]);
+                } // for w
+            }  // for h
+        } // for c
+    }  // for n
+
+}
+
+void  MMCVCornerPoolKernel::Compute(OrtKernelContext *context) {
+    const int mode = int(mode_);
+    typedef float T;
+    const OrtValue *input = ort_.KernelContext_GetInput(context, 0);
+    const T *input_data = reinterpret_cast<const float *>(ort_.GetTensorData<T>(input));
+
+    // get output memory
+    OrtTensorDimensions out_dimensions(ort_, input);
+    OrtValue *output = ort_.KernelContext_GetOutput(context, 0, out_dimensions.data(), out_dimensions.size());
+    T *output_data = ort_.GetTensorMutableData<T>(output);
+
+    // get output_size
+    int batch_size = out_dimensions.data()[0];
+    int input_channels = out_dimensions.data()[1];
+    int input_height = out_dimensions.data()[2];
+    int input_width = out_dimensions.data()[3];
+    int output_size = batch_size * input_channels * input_height * input_width;
```

Comment:
As long as output_size is used to compute batch_size inside forward function, could we just pass batch_size to the forward?